pkg/util/provider/machinecontroller/machine.go

```diff
@@ -198,10 +207,13 @@ func (c *controller) addNodeToMachine(obj interface{}) {
 
 	machine, err := c.getMachineFromNode(key)
 	if err != nil {
-		klog.Errorf("Couldn't fetch machine %s, Error: %s", key, err)
-		return
-	} else if machine == nil {
-		return
+		if err == multipleMachineMatchError {
+			klog.Errorf("Couldn't fetch machine %s, Error: %s", key, err)
+			return
+		} else if err == noMachineMatchError {
+			klog.Errorf("Couldn't fetch machine %s, Error: %s", key, err)
+			return
+		}
```

Comment:
Does this change now make sense given that the handling for the error cases is now pretty much the same?